bitcoin/_posts/0000-00-00-advent-27.md

```diff
@@ -0,0 +1,287 @@
+---
+
+
+_Welcome to day 27 of my Bitcoin Advent Calendar. You can see the previous post
+[here](/bitcoin/2021/12/23/advent-27/) or subscribe at
+[judica.org/join](https://judica.org/join) to get the new posts in your inbox_
+
+
+I know, I know.
+
+God forbid, a roadmap. People hate roadmaps. As I've noted
+[before](/blog/2021/07/02/covenants/):
+
+> **Bitcoin Eschews Roadmaps and Agendas.**
+> I provide this maxim to make clear that this document is by no means an official
+> roadmap, narrative, or prioritization. However, it is my own assessment of what
+> the current most pragmatic approach to upgrading Bitcoin is, based on my
+> understanding of the state of outstanding proposals and their interactions.
+> 
+> My priorities in producing this are to open a discussion on potential new
+> features, risk minimization, and pragmatic design for Bitcoin.
+
+If you didn't click, I definitely recommend reading [the quoted
+post](/blog/2021/07/02/covenants/) in conjunction with this one. As well as, if
+you're a first time visitor to the Advent Calendar, the preceding 26 posts.
+
+<hr>
+
+In contrast to my prior post, this roadmap is going to be less about full
+justifications and unbiased weightings and tallyings of sentiments and more just
+me spitting out a timeline we could introduce changes on. It's not a final
+answer, and in no way authoritative, but it's a launch point for a discussion
+that has to happen in some way at some point in order to advance a soft-fork.
+
+_Consider this as being posted for the sake of public review. If you disagree
+with this, let me know why! But please no attacks just for the act of discussing
+the topic of soft-fork activation[^ordo]._
+
+[^ordo]: You do you though. It'd really hurt my feelings if you called me a Smelly Weasel attacking Bitcoin.
+
+So buckle up here's how we could make Bitcoin kick ass in 2022 and beyond:
+
+# 2022: The Year of the Covenant
+
+## BIP-119 Timeline
+CheckTemplateVerify is getting close to ready to go. There are numerous
+[supporters (listed on utxos.org)](https://utxos.org/signals), few detractors,
+and a bumper crop of amazing use cases (did you read the calendar?) waiting for
+us on the other side of CTV activation. The major critiques are that we might want
+something that does 'more' than CTV itself, or to include it in a bundle of
+things.
+
+My take: we're not Gordon Gecko. Greed is _Not_ Good. CTV represents a clean, well
+contained, unproblematic upgrade that's going to deliver hella functionality in
+service of scaling, decentralization, self custody, and privacy. Let's secure
+the bag for Bitcoin users everywhere and make it happen. We can always do more,
+later, informed by what extensions we need for rapidly maturing tools like
+Sapio. CTV is also technically specified and implemented sufficiently -- a view
+I've confirmed with a couple other devs -- that it is able to be considered for
+release.
+
+### What would have to happen to release CTV?
```

Comment:
Does it make sense to add something about a signet and/or testnet release for people to start playing around with it and potentially uncovering bugs?

Reply:
How about something about alternative implementations? Can we say you anticipate support from btcd (maybe others)?

Reply:
i ran one once a while back, no one every used it... i could spin up another no issue. It's difficult since you don't *just* need signet, also some sort of faucet system.

Also regtest works more than fine... 


w.r.t. btcd how's taproot going 🥕 ...? that said CTV is really small, no new library stuff, and @roasbeef said he'd put it together :)

Reply:
Shooting to have it all up for review by the turn of the new year 😎